scripts/irods/test/resource_suite.py

```diff
@@ -1229,4 +1229,61 @@ def test_itrim_returns_on_negative_status__ticket_3531(self):
+                    os.path.dirname(logical_path), os.path.basename(logical_path), username)
+                ])[0].strip()
+
+            self.assertEqual(str(data_access_type), str(permission_value))
+
+        compound_resource = 'demoResc'
+        cache_resource = 'cacheResc'
+        archive_resource = 'archiveResc'
+        cache_hierarchy = compound_resource + ';' + cache_resource
+        archive_hierarchy = compound_resource + ';' + archive_resource
+
+        owner_user = self.user0
+        readonly_user = self.user1
+        filename = 'foo'
+        contents = 'jimbo'
+        logical_path = os.path.join(owner_user.session_collection, filename)
 
+        try:
+            # Create a data object which should appear under the compound resource.
+            owner_user.assert_icommand(['istream', 'write', logical_path], input=contents)
+            self.assertTrue(lib.replica_exists_on_resource(owner_user, logical_path, cache_resource))
+            self.assertTrue(lib.replica_exists_on_resource(owner_user, logical_path, archive_resource))
+
+            # Grant read access to another user, ensuring that the other user can see the data object.
+            owner_user.assert_icommand(['ichmod', '-r', 'read', readonly_user.username, os.path.dirname(logical_path)])
+
+            # Ensure that the read-only user has read-only permission on the data object.
+            assert_permissions_on_data_object_for_user(readonly_user.username, logical_path, 1050)
+
+            # Trim the replica on the cache resource so that only the replica in the archive remains. Replica 0 resides
+            # on the cache resource at this point.
+            owner_user.assert_icommand(['itrim', '-N1', '-n0', logical_path], 'STDOUT')
+            self.assertFalse(lib.replica_exists_on_resource(owner_user, logical_path, cache_resource))
+            self.assertTrue(lib.replica_exists_on_resource(owner_user, logical_path, archive_resource))
+
+            # As the user with read-only access, attempt to get the data object. Replica 1 resides on the archive
+            # resource, so the replica on the cache resource which results from the stage-to-cache should be number 2.
+            readonly_user.assert_icommand(['iget', logical_path, '-'], 'STDOUT', contents)
+            self.assertEqual(str(1), lib.get_replica_status(readonly_user, os.path.basename(logical_path), 1))
+            self.assertEqual(str(1), lib.get_replica_status(readonly_user, os.path.basename(logical_path), 2))
+
+            # Ensure that the user has the same permissions on the data object as before getting it.
+            assert_permissions_on_data_object_for_user(readonly_user.username, logical_path, 1050)
+
+        finally:
+            self.admin.assert_icommand(['ils', '-Al', logical_path], 'STDOUT') # Debugging
+
+            # Make sure that the data object can be removed by marking both replicas stale before removing.
```

Comment:
Is marking the replicas stale meant to guard against intermediate replicas and make the test idempotent?